### PR TITLE
C++: Fix `cpp/use-of-unique-pointer-after-lifetime-ends` FP

### PR DESCRIPTION
**Pull request by MathiasVP**
This fixed a FP that @geoffw0 spotted earlier

### CODE REVIEW

File: cpp/ql/src/Security/CWE/CWE-416/UseOfUniquePointerAfterLifetimeEnds.ql [hunk 1]
```diff
@@ -30,6 +30,7 @@ where
   outlivesFullExpr(c) and
   not c.isFromUninstantiatedTemplate(_) and
   isUniquePointerDerefFunction(c.getTarget()) and
+  not c.getActualType() instanceof BoolType and
```
Comment (jketema) [inline cpp/ql/src/Security/CWE/CWE-416/UseOfUniquePointerAfterLifetimeEnds.ql:33]: Is there a specific reason we just exclude bool types, and not just any kind of integer type?

Comment (MathiasVP) [reply cpp/ql/src/Security/CWE/CWE-416/UseOfUniquePointerAfterLifetimeEnds.ql:33]: Not really, no 🤔 Alternatively we could require that the actual type is a pointer type to be more on the safe side?

**Review by geoffw0 — DISMISSED** [review]
:+1:

**Comment by MathiasVP** [comment]
DCA was uneventful, and CI failures are unrelated. So I'll 🤠 merge this!

**Pull request merged**